src/Controller/Api/Legacy/MediaController.php

```diff
@@ -0,0 +1,394 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Controller\Api\Legacy;
+
+use FOS\RestBundle\Context\Context;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\View\View as FOSRestView;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Sonata\DatagridBundle\Pager\PagerInterface;
+use Sonata\MediaBundle\Form\Type\ApiMediaType;
+use Sonata\MediaBundle\Model\MediaInterface;
+use Sonata\MediaBundle\Model\MediaManagerInterface;
+use Sonata\MediaBundle\Provider\MediaProviderInterface;
+use Sonata\MediaBundle\Provider\Pool;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+/**
+ * Note: Media is plural, medium is singular (at least according to FOSRestBundle route generator).
+ *
+ * @deprecated since sonata-project/media-bundle 3.21, to be removed in 4.0.
```

Comment:
```suggestion
 * @deprecated since sonata-project/media-bundle 3.x, to be removed in version 4.0. Use Sonata\MediaBundle\Controller\Api\MediaController instead.
```
BTW, I think we must add the `@final` annotation.

Reply:
`@final` annotation is missing yet.